toolkit/tools/internal/timestamp/timestamp_test.go

```diff
@@ -0,0 +1,101 @@
+// Copyright Microsoft Corporation.
+// Licensed under the MIT License.
+
+package timestamp
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+)
+
+var (
+	info1 = New("tool 1", true)
+	info2 = New("tool 2", false)
+)
+
+//TestMain found in configuration_test.go.
+
+func Test_WritetoFile_range_instant(t *testing.T) {
+	info1.Start()
+	info1.RecordToFile("test step", "test action", os.Stdout)
+}
+
+func Test_WritetoFile_noRange_instant(t *testing.T) {
+	info2.Start()
+	time.Sleep(30 * time.Millisecond)
+	info2.RecordToFile("test step", "test action", os.Stdout)
+}
+
+func Test_WritetoCSV_range(t *testing.T) {
+	InitCSV("build-time.csv", true)
+	Stamp.Start()
+	time.Sleep(10 * time.Millisecond)
+	Stamp.RecordToCSV("step 1", "action 1")
+	Stamp.Start()
+	time.Sleep(20 * time.Millisecond)
+	Stamp.RecordToCSV("step 2", "action 1")
+	Stamp.Start()
+	time.Sleep(10 * time.Millisecond)
+	Stamp.RecordToCSV("step 2", "action 2")
+	Stamp.Start()
+	time.Sleep(30 * time.Millisecond)
+	Stamp.RecordToCSV("step 3", "action 1")
+}
+
+func Test_getHomeDir(t *testing.T) {
+	home, _ := os.UserHomeDir()
+	fmt.Printf("%s\n", home)
+	curr, _ := os.Getwd()
+	fmt.Printf("%s\n", curr)
+	testDir := "/home/xuanchen/repos/pod_repo/CBL-Mariner/toolkit"
+	dirLen := len(testDir)
+	fmt.Printf("%s\n", testDir[dirLen-19:])
+}
+
+func Test_WritetoCSV_noRange(t *testing.T) {
+	InitCSV("build-time.csv", false)
+	time.Sleep(10 * time.Millisecond) // extra sleep
+	time.Sleep(20 * time.Millisecond)
+	Stamp.RecordToCSV("step 1", "action 1")
+	Stamp.Start()
+	time.Sleep(20 * time.Millisecond)
+	Stamp.RecordToCSV("step 2", "action 1")
+	time.Sleep(10 * time.Millisecond) // extra sleep
+	time.Sleep(20 * time.Millisecond)
+	Stamp.RecordToCSV("step 2", "action 2")
+	Stamp.Start()
+	time.Sleep(20 * time.Millisecond)
+	Stamp.RecordToCSV("step 3", "action 1")
+}
+
+func Test_WritetoFile_noRange(t *testing.T) {
+	info2.Start()
+	time.Sleep(10 * time.Millisecond)
+	info2.RecordToFile("step 1", "action 1", os.Stdout)
+	info2.Start()
+	time.Sleep(10 * time.Millisecond)
+	info2.RecordToFile("step 2", "action 1", os.Stdout)
+	time.Sleep(20 * time.Millisecond)
+	info2.Start()
+	time.Sleep(10 * time.Millisecond)
+	info2.RecordToFile("step 2", "action 2", os.Stdout)
+	info2.Start()
+	time.Sleep(10 * time.Millisecond)
+	info2.RecordToFile("step 3", "action 1", os.Stdout)
+}
+
+func WritetoCSV(info *TimeInfo, seconds time.Duration) {
+	Stamp.Start()
+	time.Sleep(seconds * time.Millisecond)
+	Stamp.RecordToCSV("test tool", "test step")
+}
+
+func Test_filename(t *testing.T) {
+	completePath := "/home/xuanchen/repos/pod_repo/CBL-Mariner/build/timestamp/imageconfigvalidator.csv"
```

Comment:
Don't hardcode full paths to home directories.

Reply:
hardcoded path to the specific user's home directory is still present. Please remove